Update to mmenu-js v9

rabauss opened the pull request:
This PR prepares the update to mmenu-js v9.

- [x] Update to mmenu-js v9.1.6
- [x] License of Contao Academy still valid for v9?
- [x] Refactor module options to custom config table (https://github.com/dklemmt/contao_dk_mmenu/pull/69)
- [x] Validate the support of frontend module options

rabauss commented:
Finally I managed to analyze the config options and here is the result, which we need to add to the changelog:
- Remove deprecated options: 
  - columns add
  - drag
  - effects
  - fullscreen
  - keyboard navigation
  - move background
  - on click close
  - page dim
  - polyfill
  - popup
  - shadows
- Add high contrast option for theme

fritzmg commented:
> @fritzmg Can we add a hint to eikona-media.de to the Readme?

Feel free to do so :) (I mean this is not my extension anyway ;P)

contaoacademy commented:
Would it make sense to use the extension under a new name in a new repo?

fritzmg commented:
Everybody is free to fork this extension and publish it under a new package name - as long as the license allows it.

rabauss commented:
> Vielen Dank. Mein Test mit Contao 4.13 und PHP 8.1 hat wunderbar geklappt.
> 
> Folgende Anmerkungen:
> 
> **Text bei «Art des Menüs» optimieren** Vorschlag: Bestimmt, ob Untermenüs von rechts hineingleiten **(horizontal)** oder nach unten aufklappen **(vertikal)**.
> 
> **Text bei "Zeige einen kleinen Teil übergeordneter Panele"** Zeige einen kleinen Teil übergeordneter Panele (nur horizontal)
> 
> **Wäre es möglich, den Menütitel frei zu bestimmen?** Aktuell wird hier automatisch `Menü` eingetragen. Es wäre schön, wenn das einfach in der Konfiguration überschreibbar wäre.

I updated the helping hints in the back end and I added a config option for the navbar title!

fritzmg commented:
Finally merged and released as `4.0.0` :) thank you @rabauss

Aybee commented:
> I updated the helping hints in the back end and I added a config option for the navbar title!

Könntest du da bitte noch die Inserttags in den Titel reinbringen? Im Moment funktioniert sowas hier z.B. nicht.

`{{ifnlng::de}}Universal stuff{{ifnlng}}{{iflng::de}}Deutsches Zeugs{{iflng}}`

fritzmg commented:
In what way does it not work? Contao always replaces the insert tags.

Aybee commented:
The insert tag is not resolved. As you can see it above, I entered it in the `title` field and that's how it appears in the FE.

fritzmg commented:
Do you have a link to such a page?

Aybee commented:
mmenu 4.0.0, Contao 4.9.37
https://globolog.net/_dev.html
Nur diese eine Seite, die anderen sind noch alt, wird gerade überarbeitet. Ist auch noch auf Contao 4.9, weil noch auf MetaModels für 4.13 gewartet wird.

Vielleicht liegt es daran, dass ein Sprachbaustein ausgegeben wird, wenn man das Feld leer lässt - Menu, Menü...
![Bildschirmfoto am 2022-12-09 um 12 48 49](https://user-images.githubusercontent.com/1143013/206696199-456508f7-8add-4075-b250-6ade830c0b1d.png)
![Bildschirmfoto am 2022-12-09 um 12 50 25](https://user-images.githubusercontent.com/1143013/206696920-b14b607c-8086-4e52-8ce7-7c70be3d90af.png)

fritzmg commented:
Apparently this is a limitation or bug within Contao itself. Any content that is added to `$GLOBALS['TL_BODY']` will not get its insert tags replaced.

Aybee commented:
Ok, thank you. So when I want another word than 'Menu, Menü, ...". I can change the language variable.

fritzmg commented:
In version `4.0.1` the insert tags within `mmenu_default` are now also replaced.
